### PR TITLE
Introducing SuspensionInfo and suspension chain parameters

### PR DESCRIPTION
**Pull request by drsk0**
## Purpose

These are the corresponding changes for the newly introduced validator suspension chain parameters in the node repo. Also a data structure `SuspensionInfo` is introduced that gathers the missed rounds and whether a validator is primed for suspension at the next snapshot.

## Changes

See above

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

**Review by td202 — COMMENTED** [review]
Base submodule requires updating.

File: concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs
Comment (td202) [inline concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs:1]: It looks like updating the new parameter is missing from `processUpdateQueues`. Please double check that everything that is done for the other queues is also done for the new one.

File: concordium-consensus/src/Concordium/GlobalState/PoolRewards.hs [hunk 1]
```diff
@@ -24,8 +46,8 @@ data BakerPoolRewardDetails (av :: AccountVersion) = BakerPoolRewardDetails
       transactionFeesAccrued :: !Amount,
       -- | Whether the pool contributed to a finalization proof in the reward period
       finalizationAwake :: !Bool,
-      -- | The number of missed rounds in the reward period
-      missedRounds :: !(Conditionally (SupportsValidatorSuspension av) Word64)
+      -- | Information for deciding whether a validator will be suspended the next snapshot epoch.
```
Comment (td202) [inline concordium-consensus/src/Concordium/GlobalState/PoolRewards.hs:49]:
```suggestion
      -- | Information for deciding whether a validator will be suspended at the next snapshot epoch.
```

**Review by td202 — CHANGES_REQUESTED** [review]

File: concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs
Comment (drsk0) [reply concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs:1]: I did another grep `FinalizationCommitteeParameters`, it seems all the boilerplate should be there now. However, I forgot to add the new chain parameter version in the grpc api: https://github.com/Concordium/concordium-grpc-api/pull/73.

**Review by td202 — APPROVED** [review]
Looks good!

**Pull request merged**